### PR TITLE
Build CoreCLR projects into their own directory

### PR DESCRIPTION
**Pull request by davkean**
Fixes: #6423.

Also turn on hardlinks where possible, so we try to reduce drop size given we're duplicating lots of binaries.

### CODE REVIEW

**Comment by davkean** [comment]
tag @dotnet/roslyn-infrastructure @jaredpar @agocke

**Comment by amcasey** [comment]
FYI @tmat

File: src/Compilers/CSharp/CscCore/CscCore.csproj [hunk 1]
```diff
@@ -15,7 +15,7 @@
     <LargeAddressAware>true</LargeAddressAware>
     <StartupObject>Microsoft.CodeAnalysis.CSharp.CommandLine.Program</StartupObject>
     <SolutionDir Condition="'$(SolutionDir)' == '' OR '$(SolutionDir)' == '*Undefined*'">..\..\..\..\</SolutionDir>
-    <OutDir>$(OutDir)core-clr\</OutDir>
+    <OutDir>$(OutDir)csccore\</OutDir>
```
Comment (jaredpar) [inline src/Compilers/CSharp/CscCore/CscCore.csproj]: Can we have a root directory name coreclr and have the projects build individuually under them?

File: build/Targets/VSL.Settings.targets [hunk 2]
```diff
@@ -11,6 +11,9 @@
                                  '$(OS)' != 'Windows_NT'">$([System.Environment]::GetFolderPath(SpecialFolder.Personal))\.nuget\packages</NuGetPackageRoot>
     <BaseNuGetRuntimeIdentifier Condition="'$(OS)' == 'Windows_NT'">win7</BaseNuGetRuntimeIdentifier>
     <BaseNuGetRuntimeIdentifier Condition="'$(OS)' == 'Unix'">ubuntu.14.04</BaseNuGetRuntimeIdentifier>
+
+    <!-- Turn on hard-linking on Windows to reduce copy time and disk space -->
+    <CreateHardLinksForCopyLocalIfPossible Condition="'$(OS)' == 'Windows_NT'">true</CreateHardLinksForCopyLocalIfPossible>
```
Comment (jaredpar) [inline build/Targets/VSL.Settings.targets]: Does this mean we'll still have bad perf and huge binary directories on Unix?

**Pull request merged**